elf_reader.go

```diff
@@ -1067,6 +1094,96 @@ func (ec *elfCode) loadDataSections(maps map[string]*MapSpec) error {
 
 		maps[sec.Name] = mapSpec
 	}
+
+	if ec.btf != nil {
+		var ds *btf.Datasec
+		if err := ec.btf.TypeByName(".kconfig", &ds); err != nil {
+			if errors.Is(err, btf.ErrNotFound) {
+				return nil
+			}
+			return err
+		}
+
+		maps[".kconfig"] = &MapSpec{
+			Name:       ".kconfig",
+			Type:       Array,
+			KeySize:    uint32(4),
+			MaxEntries: 1,
+			Flags:      unix.BPF_F_RDONLY_PROG | unix.BPF_F_MMAPABLE,
+			Key:        &btf.Void{},
+			Value:      ds,
+			ValueSize:  ds.Size,
+		}
+	}
+
+	return nil
+}
+
+func (ec *elfCode) populateDataSections() error {
+	if ec.maps == nil {
+		return fmt.Errorf("elfCode map of MapSpec should not be nil")
+	}
+
+	return ec.populateKconfigSection()
+}
+
+func (ec *elfCode) populateKconfigSection() error {
+	mapSpec, ok := (*ec.maps)[".kconfig"]
+	if !ok {
+		// If .kconfig map does not exist it just means the code does not have any
+		// variable inside .kconfig data section.
+		return nil
+	}
+
+	ds, ok := mapSpec.Value.(*btf.Datasec)
+	if !ok {
+		return fmt.Errorf(".kconfig value must be of type btf.Datasec")
+	}
+
+	data := make([]byte, 0, ds.Size)
+	for _, vsi := range ds.Vars {
+		v, ok := vsi.Type.(*btf.Var)
+		if !ok {
+			return fmt.Errorf("info should point to a variable")
+		}
+
+		s, err := btf.Alignof(v.Type)
+		if err != nil {
+			return fmt.Errorf("getting size of kconfig variable %s: %w", v.TypeName(), err)
```

Comment:
For consistency with the above:
```suggestion
			return fmt.Errorf("variable %s: getting size: %w", v.Name, err)
```